src/StoreForwarding.cpp

```diff
@@ -1222,17 +1223,17 @@ class RenameVars : public IRMutator {
     }
 };
 
-vector<Store> find_non_aliasing_stores(Stmt stmt) {
+vector<const Store *> find_non_aliasing_stores(Stmt stmt) {
```

Comment:
Yes, I did consider the danger, but the problem was that doing this resulted in this error.
"error: call to implicitly-deleted copy constructor of 'Halide::Internal::Store"

I realized that this is because the copy constructor for atomic that is required by the reference count is explicitly deleted.

/usr/bin/../lib/gcc/x86_64-linux-gnu/4.9/../../../../include/c++/4.9/atomic:605:7: note: 'atomic' has been explicitly marked deleted here
      atomic(const atomic&) = delete;

I think I should just write a copy constructor for the Halide::Internal::Store?